proof/verifier_test.go

```diff
@@ -0,0 +1,613 @@
+			// root commitment changed, and we don't update the
+			// anchor tx output yet.
+			_, _ = addV1InclusionProof(
+				t, p, rootCommitment, withAddStxoProof(),
+			)
+
+			return p
+		},
+		expectedErrContains: "error verifying v0 inclusion proof",
+	}, {
+		name: "missing v1 proof",
+		makeProof: func(t *testing.T) *Proof {
+			p, rootCommitment := makeV0InclusionProof(t)
+			p.Version = TransitionV1
+
+			// But now we update the anchor tx output, which should
+			// cause the proof to be invalid, since we didn't add
+			// the STXO asset to the root commitment yet.
+			_, _ = addV1InclusionProof(
+				t, p, rootCommitment, withUpdateAnchorTx(),
+			)
+
+			return p
+		},
+		expectedErr:         ErrInvalidCommitmentProof,
+		expectedErrContains: "missing STXO proofs",
+	}, {
+		name: "correct v0 and v1 proofs",
+		makeProof: func(t *testing.T) *Proof {
+			p, rootCommitment := makeV0InclusionProof(t)
+			p.Version = TransitionV1
+
+			// We add an STXO asset to the root commitment, and we
+			// also update the anchor TX.
+			_, _ = addV1InclusionProof(
+				t, p, rootCommitment, withAddStxoProof(),
+				withUpdateAnchorTx(),
+			)
+
+			return p
+		},
+	}, {
+		name: "invalid v1 proof",
+		makeProof: func(t *testing.T) *Proof {
+			p, rootCommitment := makeV0InclusionProof(t)
+			p.Version = TransitionV1
+
+			// We add an STXO asset to the root commitment.
+			_, key := addV1InclusionProof(
+				t, p, rootCommitment, withAddStxoProof(),
+				withUpdateAnchorTx(),
+			)
+
+			// At this point both proofs are valid. But we want to
+			// invalidate just the v1 proof, so we replace it by one
+			// that is derived from a different commitment. We can
+			// achieve that by downgrading the commitment and
+			// re-deriving the same proof, without updating the
+			// anchor output transaction.
+			clonedCommitment, err := rootCommitment.Downgrade()
+			require.NoError(t, err)
+
+			_, v1ProofOld, err := clonedCommitment.Proof(
+				asset.EmptyGenesisID,
+				p.Asset.AssetCommitmentKey(),
+			)
+			require.NoError(t, err)
+
+			cp := p.InclusionProof.CommitmentProof
+			cp.STXOProofs[key] = *v1ProofOld
+
+			return p
+		},
+		expectedErrContains: "error verifying STXO proof: invalid " +
+			"taproot proof",
+	}, {
+		name: "stxo proof missing",
+		makeProof: func(t *testing.T) *Proof {
+			// We create a proof for an asset that spends two
+			// inputs.
+			moreInputs := []asset.Witness{{
+				PrevID: &asset.PrevID{
+					ID: asset.RandID(t),
+				},
+			}}
+			p, rootCommitment := makeV0InclusionProof(
+				t, withAdditionalPrevWitnesses(moreInputs),
+			)
+			p.Version = TransitionV1
+
+			// We add the first STXO asset to the root commitment.
+			// The second one is missing, so the proof should be
+			// invalid.
+			_, _ = addV1InclusionProof(
+				t, p, rootCommitment, withAddStxoProof(),
+				withUpdateAnchorTx(),
+			)
+
+			return p
+		},
+		expectedErr:         ErrInvalidCommitmentProof,
```

Comment:
Nice set of tests! I think for the sake of extneding the test vectors here in the future, we should make some new specific errors for these sub-cases. So in this case, it would be like `ErrStxoInputProofMissing` or something like that.